build_sysext

```diff
@@ -0,0 +1,94 @@
+# found in the LICENSE file. Script to generate sysext. See systemd-sysext(8).
+# Prerequisite that you've run build_packages and build_image
+
+SCRIPT_ROOT=$(dirname "$(readlink -f "$0")")
+. "${SCRIPT_ROOT}/common.sh" || exit 1
+
+# Script must run inside the chroot
+assert_inside_chroot
+assert_root_user
+
+DEFINE_string board "${DEFAULT_BOARD}" \
+  "The board to build an image for."
+
+FLAGS_HELP="USAGE: build_sysext [flags] [package to build].
+This script is used to build a Flatcar sysext. 
+
+Examples:
+
+build_sysext --board=amd64 python - builds python sysext for amd64 architecture.
+
+"
+
+show_help_if_requested "$@"
+
+# Parse command line
+FLAGS "$@" || exit 1
+
+eval set -- "${FLAGS_ARGV:-prod}"
+
+. "${BUILD_LIBRARY_DIR}/release_util.sh" || exit 1
+
+. "${BUILD_LIBRARY_DIR}/toolchain_util.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/board_options.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/build_image_util.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/prod_image_util.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/dev_container_util.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/test_image_content.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/torcx_manifest.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/vm_image_util.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/cros_vm_constants.sh" || exit 1
+
+BUILD_DIR="/mnt/host/source/src/build/images/amd64-usr/developer-3621.0.0+nightly-20230531-2100-66-gf786da53f9-a1"
+
+echo "${BUILD_DIR}"
+
+cleanup() {
+  rm -rf "${BUILD_DIR}/fs-root" "${BUILD_DIR}/install-root" "${BUILD_DIR}/workdir" || exit 1
+  umount "${BUILD_DIR}/fs-root" "${BUILD_DIR}/install-root" "${BUILD_DIR}/workdir" || exit 1
+  rm "${BUILD_DIR}/${SYSEXTNAME}.raw"
+}
+
+# Set up trap to execute cleanup() on script exit
+
+# trap cleanup EXIT
+
+export SYSEXTNAME=$1
+
+case $BOARD in
+*arm64-usr*) ARCH=arm64 ;;
+*amd64-usr*) ARCH=x86-64 ;;
+*) echo "Unknown board" ;;
+esac
+
+source /usr/lib/os-release
+
+umount "${BUILD_DIR}/fs-root" "${BUILD_DIR}/install-root" "${BUILD_DIR}/workdir" || true
+
+mkdir "${BUILD_DIR}/fs-root"
+mount -rt squashfs -o loop,nodev,noexec "${BUILD_DIR}/flatcar_production_image_sysext.squashfs" "${BUILD_DIR}/fs-root"
+mkdir "${BUILD_DIR}/install-root"
+mkdir "${BUILD_DIR}/workdir"
+mount -t overlay overlay -o lowerdir="${BUILD_DIR}/fs-root",upperdir="${BUILD_DIR}/install-root",workdir="${BUILD_DIR}/workdir" "${BUILD_DIR}/install-root"
+VERSION_BOARD=$(grep "^VERSION=" ${BUILD_DIR}/fs-root/usr/lib/os-release | cut -d = -f 2-)
+if [ "$VERSION_BOARD" != "$VERSION" ]; then
+  echo "$VERSION_BOARD"
+  echo "$VERSION"
+  echo "Version mismatch between board flatcar release and SDK container flatcar release"
+  exit 1
+else
+  /usr/local/bin/emerge-${BOARD} --jobs=${NUM_JOBS} --getbinpkg "${SYSEXTNAME}"
+  ROOT="${BUILD_DIR}/install-root" PORTAGE_CONFIGROOT="/build/${BOARD}" FEATURES="-ebuild-locks" emerge --root-deps=rdeps --usepkgonly --verbose "${SYSEXTNAME}"
+  umount "${BUILD_DIR}/install-root" || true
+  umount "${BUILD_DIR}/fs-root" || true
```

Comment:
Wouldn't be needed if `cleanup` is used at the start

Reply:
Ah, no, you umount because you want to get rid of the overlay, please add a comment.